relay-auth/src/lib.rs

```diff
@@ -148,6 +148,16 @@ pub enum UnpackError {
     SignatureExpired,
 }
 
+/// Used to tell which algorithm was used for signature creation.
+#[derive(Default, Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum SignatureAlgorithm {
+    /// Regular signature creation which clones the data internally.
+    #[default]
```

Comment:
Not sure, but I am kinda leaning towards not having a default but instead making the field optional, just to indicate that `Regular` shouldn't be the default it was just what we started out with.

Ideally everything uses the sha version imo.